### PR TITLE
Iterative eigendecomposition

### PR DESCRIPTION
**Pull request by jrs65**
Not quite finished, but I'd like to get people to start looking over it.

- Big change is the new iterative decomposition process.
- Minor change is changing how visWriter tracks the last update of acqs to make it possible to replay.

Tagging @kiyo-masui to look at the eigenvalue algorithm, @nritsche to look at the change to the last update tracking in visWriter and tagging @andrerenard for an overall look.

Still to come:
- Changes to the baseline masking to remove whole cylinders.
- Back port this to apply it to master. Currently built from develop (so I need to swap stage -> process etc)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,6 +30,7 @@ Cmake build options:
 * `-DUSE_FFTW=ON` - Build an FFTW-based F-engine. Requires FFTW3.
 * `-DUSE_LAPACK=ON` - Build processes depending on LAPACK. Currently only OpenBLAS built from source is supported (see above).
 * `-DOPENBLAS_PATH=<openblas_prefix>` - Path to OpenBLAS installation, if not in the `CMAKE_PREFIX_PATH`
+* `-DBLAZE_PATH=<blaze_path>` - Path to BLAZE headers.
```
Comment (andrerenard) [inline README.md]: You might want to add that this requires `-DUSE_LAPACK=ON`

**Comment by jrs65** [comment]
Thanks @andrerenard I bunch of these are issues in the code I based off from `eigenVis`. I'll try and fix them up.

File: lib/processes/EigenVisIter.cpp [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+        output_frame.erms = stats.converged ? stats.rms : -stats.eps_eval;
+
+        // Finish up interation.
+        mark_frame_empty(input_buffer, unique_name.c_str(), input_frame_id++);
+        mark_frame_full(output_buffer, unique_name.c_str(), output_frame_id++);
+    }
+}
+
+
+// TODO: this should probably be a private method, but that would mean exposing all the blaze stuff and [j]
```
[the remark this answers is not in the document]
Comment (jrs65) [reply lib/processes/EigenVisIter.cpp]: Haha. Yeah, that's an orphaned comment. Originally it wasn't a private method, just a static function, but then I changed my mind.

**Comment by nritsche** [comment]
What do you mean by "data replay"?

**Comment by jrs65** [comment]
@nritsche I mean the ability to take historic data and process it through kotekan at a later time. In practice this consists of a few parts:
1. A Python script (not in this PR) that takes data from andata and writes out a vis raw file of the data.
2. A reader in kotekan for this. `visRawReader` does this with essentially no modifications.
3. The ability to write out the replayed data. Because `visWriter` checks if an acquisition it is writing is stale and closes it, this required a change to make that check be based on the clock time that it last saw data in that acquisition, and *not* the actual timestamp of the data (which may be months ago when replaying data). That change I have made in this PR.

**Comment by jrs65** [comment]
@andrerenard I should have finished up fixing your comments. I'm going to try running on the test nodes this afternoon.

**Comment by jrs65** [comment]
@andrerenard it looks like the CI server is going to need to have Blaze installed. Just checking out the repo: https://bitbucket.org/blaze-lib/blaze and pointing `BLAZE_PATH` at it should be sufficient.

**Comment by jrs65** [comment]
Also the worst thing I'm going to do in the next hour is change a few config parameters. So I think you can start merging things if you're ready to.

**Comment by nritsche** [comment]
> @andrerenard it looks like the CI server is going to need to have Blaze installed. Just checking out the repo: https://bitbucket.org/blaze-lib/blaze and pointing `BLAZE_PATH` at it should be sufficient.

@andrerenard I did this, it looks like the CI server is not getting triggered by the commit.

**Comment by andrerenard** [comment]
@nritsche yeah something is wrong with the CI server triggering. I've just been manually triggering it until I have some time to find a solution.

File: lib/processes/EigenVisIter.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (andrerenard) [reply lib/processes/EigenVisIter.cpp]: It looks like this comment didn't get removed.

**Pull request merged**